crates/bindings-macro/src/view.rs

```diff
@@ -0,0 +1,284 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::parse::Parser;
+use syn::{FnArg, ItemFn};
+
+use crate::sym;
+use crate::util::{ident_to_litstr, match_meta};
+
+pub(crate) struct ViewArgs {
+    anonymous: bool,
+}
+
+impl ViewArgs {
+    /// Parse `#[view(public)]` where public is required
+    pub(crate) fn parse(input: TokenStream) -> syn::Result<Self> {
+        if input.is_empty() {
+            return Err(syn::Error::new(
+                proc_macro2::Span::call_site(),
+                "views must be declared as #[view(public)]; public is required",
+            ));
+        }
+        let mut public = false;
+        let mut anonymous = false;
+        syn::meta::parser(|meta| {
+            match_meta!(match meta {
+                sym::public => {
+                    public = true;
+                }
+                sym::anonymous => {
+                    anonymous = true;
+                }
+            });
+            Ok(())
+        })
+        .parse2(input)?;
+        if !public {
+            return Err(syn::Error::new(
+                proc_macro2::Span::call_site(),
+                "views must be declared as #[view(public)]; public is required",
+            ));
+        }
+        Ok(Self { anonymous })
+    }
+}
+
+fn view_impl_anon(original_function: &ItemFn) -> syn::Result<TokenStream> {
+    let func_name = &original_function.sig.ident;
+    let view_name = ident_to_litstr(func_name);
+    let vis = &original_function.vis;
+
+    for param in &original_function.sig.generics.params {
+        let err = |msg| syn::Error::new_spanned(param, msg);
+        match param {
+            syn::GenericParam::Lifetime(_) => {}
+            syn::GenericParam::Type(_) => return Err(err("type parameters are not allowed on views")),
+            syn::GenericParam::Const(_) => return Err(err("const parameters are not allowed on views")),
+        }
+    }
+
+    // Extract all function parameters, except for `self` ones that aren't allowed.
+    let typed_args = original_function
+        .sig
+        .inputs
+        .iter()
+        .map(|arg| match arg {
+            FnArg::Typed(arg) => Ok(arg),
+            FnArg::Receiver(_) => Err(syn::Error::new_spanned(arg, "`self` arguments not allowed in views")),
+        })
+        .collect::<syn::Result<Vec<_>>>()?;
+
+    // Extract all function parameter names.
+    let opt_arg_names = typed_args.iter().map(|arg| {
+        if let syn::Pat::Ident(i) = &*arg.pat {
+            let name = i.ident.to_string();
+            quote!(Some(#name))
+        } else {
+            quote!(None)
+        }
+    });
+
+    let arg_tys = typed_args.iter().map(|arg| arg.ty.as_ref()).collect::<Vec<_>>();
+
+    // Extract the context type
+    let ctx_ty = arg_tys.first().ok_or_else(|| {
```

Comment:
Could use `let [ctx_ty, arg_tys @ ..] = arg_tys else { ... }` here to avoid `.skip(1)` later.